## Black-Scholes-Merton

<b>What is the Black-Scholes-Merton Model?</b>
- The Black-Scholes-Merton (BSM) model is a pricing model for financial instruments. It is used for the valuation of stock options. The BSM model is used to determine the fair prices of stock options based on six variables: volatility, type, underlying stock price, strike price, time, and risk-free rate. It is based on the principle of hedging and focuses on eliminating risks associated with the volatility of underlying assets and stock options.

<b>Assumptions of the Black-Scholes-Merton Model</b>
- <b>Lognormal distribution:</b> The Black-Scholes-Merton model assumes that stock prices follow a lognormal distribution based on the principle that asset prices cannot take a negative value; they are bounded by zero.
- <b>No dividends:</b> The BSM model assumes that the stocks do not pay any dividends or returns.
- <b>Expiration date:</b> The model assumes that the options can only be exercised on its expiration or maturity date. Hence, it does not accurately price American options. It is extensively used in the European options market.
- <b>Random walk:</b> The stock market is a highly volatile one, and hence, a state of random walk is assumed as the market direction can never truly be predicted.
- <b>Frictionless market:</b> No transaction costs, including commission and brokerage, is assumed in the BSM model.
- <b>Risk-free interest rate:</b> The interest rates are assumed to be constant, hence making the underlying asset a risk-free one.
- <b>Normal distribution:</b> Stock returns are normally distributed. It implies that the volatility of the market is constant over time.
- <b>No arbitrage:</b> There is no arbitrage. It avoids the opportunity of making a riskless profit.
 

<b>The formula</b>
![bsf_compressed.jpg](bsf_compressed.jpg)

$Nd_{2}$ - is the probability that the call will be exercised, so $$\frac{E}{e^{rT}}\cdot Nd_{2} $$ is what you expect to pay if you exercise the option, discounted to today.<br>
In the equation $$S\cdot Nd_{1}$$ is what you can expect to receive from selling the stock, if the option has been exercised, also discounted to today.

$d_{1}$ and $d_{2}$ depend on the assumptions we have made about how the stock price evolves over time, the elements in the option contract (the stock price, exercise price and time to maturity) and the other inputs – the risk-free rate and the volatility of returns. The probabilities in the Black-Scholes model are functions of $d_{1}$ and $d_{2}$ <br>
If you know $d_{1}$ and $d_{2}$, then you can find out what $N d_{1}$ and $N d_{2}$ are from the standard normal distribution table (these are the probabilities corresponding to observing values less than $d_{1}$ and $d_{2}$ , respectively). With these probabilities you can then use the Black-Scholes model to obtain the option value, $C$

<b>Phase 1:</b>
- Import the necessary libraries
- Download the data for Microsoft (‘MSFT’) from Yahoo Finance for the period ‘2000-1-1’ until today.

In [1]:
import numpy as np
import pandas as pd
import yfinance as yf
from scipy.stats import norm

ticker = 'MSFT'  
data = pd.DataFrame()  
data[ticker] = yf.download(ticker, start='2000-1-1')['Adj Close']

[*********************100%***********************]  1 of 1 completed


In [2]:
data.tail()

MSFT
Date                  
2021-06-07  253.809998
2021-06-08  252.570007
2021-06-09  253.589996
2021-06-10  257.239990
2021-06-11  257.890015

<b>Phase 2: </b>
- Define functions for $d_{1}$, $d_{2}$ and BSM (also referred as Black Scholes Model)

In [3]:
def d1(S, K, r, stdev, T):
    return (np.log(S / K) + (r + stdev ** 2 / 2) * T) / (stdev * np.sqrt(T))
 
def d2(S, K, r, stdev, T):
    return (np.log(S / K) + (r - stdev ** 2 / 2) * T) / (stdev * np.sqrt(T))

def BSM(S, K, r, stdev, T):
        return (S * norm.cdf(d1(S, K, r, stdev, T))) - (K * np.exp(-r * T) * norm.cdf(d2(S, K, r, stdev, T)))

Store the annual standard deviation of the log returns in a variable, called “stdev”.

In [4]:
log_returns = np.log(data/data.shift(1))

In [5]:
stdev = log_returns.std() * 250**0.5

In [6]:
stdev

MSFT    0.306666
dtype: float64

<b> Phase 3: </b>
- Define r, K & T
- Call the functions d1, d2, & BSM passing the appropriate parameters

Set the risk free rate, r, equal to 2.5% (0.025); the strike price, K, equal to 280.0; and the time horizon, T, equal to 1, respectively.

In [7]:
r = 0.025
K = 280.0
T = 1

Create a variable S equal to the last adjusted closing price of Microsoft. Use the “iloc” method.

In [8]:
S = data.iloc[-1]

Call the d1 and d2 functions with the relevant arguments to obtain their values.

In [9]:
d1(S, K, r, stdev, T)

MSFT   -0.033373
dtype: float64

In [10]:
d2(S, K, r, stdev, T)

MSFT   -0.340039
dtype: float64

Use the BSM function to estimate the price of a call option, given you know the values of S, K, r, stdev, and T.

In [11]:
BSM(S, K, r, stdev, T)

MSFT    25.312882
Name: 2021-06-11 00:00:00, dtype: float64

<b>Verifying Results with an online calculator</b>

![BSM.png](BSM.png)

- <b>The calculated value of BSM 25.31 roughly matches with the value of online calculator i.e 25.16 </b>
- Hence, the model is producing reliable results 
- Source - https://zerodha.com/tools/black-scholes/

<b>Limitations of the Black-Scholes-Merton Model</b>
- <b>Limited to the European market:</b> As mentioned earlier, the Black-Scholes-Merton model is an accurate determinant of European option prices. It does not accurately value stock options in the US. It is because it assumes that options can only be exercised on its expiration/maturity date.<br><br>
- <b>Risk-free interest rates:</b> The BSM model assumes constant interest rates, but it is hardly ever the reality.<br><br>
- <b>Assumption of a frictionless market:</b> Trading generally comes with transaction costs such as brokerage fees, commission, etc. However, the Black Scholes Merton model assumes a frictionless market, which means that there are no transaction costs. It is hardly ever the reality in the trading market.<br><br>
- <b>No returns:</b> The BSM model assumes that there are no returns associated with the stock options. There are no dividends and no interest earnings. However, it is not the case in the actual trading market. The buying and selling of options are primarily focused on the returns.

References:
- https://www.futurelearn.com/info/courses/risk-management/0/steps/39298
- https://corporatefinanceinstitute.com/resources/knowledge/trading-investing/black-scholes-merton-model/
